pkg/apis/resource/validation/validation.go

```diff
@@ -372,18 +393,29 @@ func validateResourceClaimStatusUpdate(status, oldStatus *resource.ResourceClaim
 		func(consumer resource.ResourceClaimConsumerReference) (types.UID, string) { return consumer.UID, "uid" },
 		fldPath.Child("reservedFor"))...)
 
-	var allocatedDevices sets.Set[structured.DeviceID]
+	var allocatedDevices sets.Set[structured.SharedDeviceID]
 	if status.Allocation != nil {
 		allocatedDevices = gatherAllocatedDevices(&status.Allocation.Devices)
 	}
-	allErrs = append(allErrs, validateSet(status.Devices, -1,
-		func(device resource.AllocatedDeviceStatus, fldPath *field.Path) field.ErrorList {
-			return validateDeviceStatus(device, fldPath, allocatedDevices)
-		},
-		func(device resource.AllocatedDeviceStatus) (structured.DeviceID, string) {
-			return structured.MakeDeviceID(device.Driver, device.Pool, device.Device), "deviceID"
-		},
-		fldPath.Child("devices"))...)
+	validateItem := func(device resource.AllocatedDeviceStatus, fldPath *field.Path) field.ErrorList {
+		return validateDeviceStatus(device, fldPath, allocatedDevices)
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAConsumableCapacity) || shareIDInUse(status) {
+		allErrs = append(allErrs, validateSet(status.Devices, -1,
+			validateItem,
+			func(device resource.AllocatedDeviceStatus) (structured.SharedDeviceID, string) {
+				deviceID := structured.MakeDeviceID(device.Driver, device.Pool, device.Device)
+				return structured.MakeSharedDeviceID(deviceID, (*types.UID)(device.ShareID)), "sharedDeviceID"
```

Comment:
the existing code is already a little weird, returning not-actual field paths names for the id field, let's at least keep the synthetic field path identical for both paths (will require tweaking a couple unit tests)

```suggestion
				return structured.MakeSharedDeviceID(deviceID, (*types.UID)(device.ShareID)), "deviceID"
```

Reply:
@liggitt  Thank you. In that case, I believe that there is no need to separate the case since I can use MakeSharedDeviceID for both. Please see [this change](https://github.com/kubernetes/kubernetes/compare/5174c6213d09829f389086e89d5587ec75a387d9..545a167a67a9de5d2639025d527136131b8791f8).

Accordingly, shareIDInUse is removed in [this chanage](https://github.com/kubernetes/kubernetes/compare/9ff12c1bfe009a960adeb00b993b9ff02782451a..2c3f9f12f84b1963358d2152960ddb5d4a5d30bf)